app/assets/javascripts/class_lists/ClassListCreatorPage.js

```diff
@@ -462,22 +509,30 @@ export default class ClassListCreatorPage extends React.Component {
         onPrincipalNoteChanged={this.onPrincipalNoteChanged}
         onFeedbackTextChanged={this.onFeedbackTextChanged}
         onSubmitClicked={this.onSubmitClicked}
+        onClassListsChangedByPrincipal={this.onClassListsChangedByPrincipal}
+        onPrincipalTeacherNamesByRoomChanged={this.onPrincipalTeacherNamesByRoomChanged}
       />
     );
   }
 }
 ClassListCreatorPage.propTypes = {
+  currentEducator: React.PropTypes.shape({
+    id: React.PropTypes.number.isRequired,
+    admin: React.PropTypes.bool.isRequired,
+    school_id: React.PropTypes.number,
+    labels: React.PropTypes.arrayOf(React.PropTypes.string).isRequired
+  }).isRequired,
   defaultWorkspaceId: React.PropTypes.string,
   disableHistory: React.PropTypes.bool,
   disableSizing: React.PropTypes.bool,
   autoSaveIntervalMs: React.PropTypes.number
 };
 ClassListCreatorPage.defaultProps = {
-  autoSaveIntervalMs: 5000
+  autoSaveIntervalMs: 1000
```

Comment:
Just curious, why are we increasing the autosave interval to once per second here? That seems like a lot of saving and a lot of hitting the server.

Reply:
Good catch!  For teaching teams, one thing that came up is folks didn't understand if their work was saving.  We could add a message about that, but the natural space in the horizontal bar showing the steps is too crowded to fit it.  For principals there's also no final "submit" step in the same way to be like "I am done."  So I just reduced the interval here.

In terms of the number of requests, the variable name here is probably misleading - the value is used to throttle how often calls to the server are made, not to make a call to the server every 1000ms.  So on PD day when teaching teams were all doing this at once, it could increase the volume of requests to the server.  But on that day we were only at 1/3rd of the volume we had planned for, so there's plenty of headroom still.  The place the interval most matters is for text fields when folks are doing sustained typing.  There the throttling really would reduce the number of requests quite a bit (eg, if you type straight for 10 seconds this will make ~10 requests whereas before it'd make ~2).  But for principals there's no place they're doing sustained typing, and so the rate they're making changes will be lower anyway.

So I think this will be okay, but really the better thing would be to make the saving state more explicit.  We might have time to do that next week but probably not today.